tests/countries/test_cook_islands.py

```diff
@@ -0,0 +1,335 @@
+            "2012-06-06",
+            "2020-07-03",
+            "2021-07-02",
+            "2022-07-01",
+            "2023-07-07",
+            "2024-07-05",
+            "2025-07-04",
+        )
+        self.assertHolidayName(name, range(2012, 2050))
+        self.assertNoHolidayName(name, range(2000, 2012))
+
+    def test_constitution_day(self):
+        name = "Constitution Day"
+        self.assertHolidayName(name, (f"{year}-08-04" for year in range(2000, 2050)))
+        obs_dt = (
+            "2012-08-06",
+            "2013-08-05",
+            "2018-08-06",
+            "2019-08-05",
+            "2024-08-05",
+        )
+        self.assertHolidayName(f"{name} (observed)", obs_dt)
+        self.assertNoNonObservedHoliday(obs_dt)
+
+    def test_cook_islands_gospel_day(self):
+        name = "Cook Islands Gospel Day"
+        self.assertHolidayName(name, (f"{year}-10-26" for year in range(2000, 2050)))
+        obs_dt = (
+            "2013-10-28",
+            "2014-10-27",
+            "2019-10-28",
+            "2024-10-28",
+            "2025-10-27",
+        )
+        self.assertHolidayName(f"{name} (observed)", obs_dt)
+
+    def test_christmas_day(self):
+        name = "Christmas Day"
+        self.assertHolidayName(name, (f"{year}-12-25" for year in range(2000, 2050)))
+        obs_dt = (
+            "2010-12-27",
+            "2011-12-27",
+            "2016-12-27",
+            "2021-12-27",
+            "2022-12-27",
+        )
+        self.assertHolidayName(f"{name} (observed)", obs_dt)
+        self.assertNoNonObservedHoliday(obs_dt)
+
+    def test_boxing_day(self):
+        name = "Boxing Day"
+        self.assertHolidayName(name, (f"{year}-12-26" for year in range(2000, 2050)))
+        obs_dt = (
+            "2009-12-28",
+            "2010-12-28",
+            "2015-12-28",
+            "2020-12-28",
+            "2021-12-28",
+        )
+        self.assertHolidayName(f"{name} (observed)", obs_dt)
+        self.assertNoNonObservedHoliday(obs_dt)
+
+    def test_penrhyn_gospel_day(self):
+        name = "Penrhyn Gospel Day"
+        self.assertHolidayName(name, (f"{year}-03-13" for year in range(2000, 2012)))
+        obs_dt = (
+            "2004-03-15",
+            "2005-03-14",
+            "2010-03-15",
+            "2011-03-14",
+        )
+        self.assertHolidayName(f"{name} (observed)", obs_dt)
+        self.assertNoHolidayName(name, range(2012, 2050))
+        self.assertNoHolidayName(f"{name} (observed)", range(2012, 2050))
+
+    def test_palmerston_gospel_day(self):
+        name = "Palmerston Gospel Day"
+        self.assertHolidayName(name, (f"{year}-05-25" for year in range(2000, 2012)))
+        obs_dt = (
+            "2002-05-27",
+            "2003-05-26",
+            "2008-05-26",
+        )
+        self.assertHolidayName(f"{name} (observed)", obs_dt)
+        self.assertNoHolidayName(name, range(2012, 2050))
+        self.assertNoHolidayName(f"{name} (observed)", range(2012, 2050))
```

Comment:
```suggestion
        self.assertNoHolidayName(name, range(2012, 2050))
        obs_dt = (
            "2004-03-15",
            "2005-03-14",
            "2010-03-15",
            "2011-03-14",
        )
        name_observed = f"{name} (observed)"
        self.assertHolidayName(name_observed, obs_dt)
        self.assertNoNonObservedHoliday(obs_dt)
        self.assertNoHolidayName(name_observed, range(2012, 2050))
```
And same for the rest, except Atiu and Mitiaro; for them
`        self.assertNoNonObservedHolidayName(name_observed, obs_dt)
`
instead of
`        self.assertNoNonObservedHoliday(obs_dt)
`